wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ViewModel/ConversationListViewControllerViewModel.swift

```diff
@@ -190,16 +193,41 @@ extension ConversationListViewController.ViewModel {
         return true
     }
 
+    private func updateE2EICertifiedStatus() {
+        guard let contextProvider = userSession as? ContextProvider else { return }
+        Task {
+            do {
+                let context = contextProvider.syncContext
+                let (selfUser, selfConversation) = await context.perform {
+                    let selfUser = ZMUser.selfUser(in: context)
+                    let mlsSelfConversation = ZMConversation.fetchSelfMLSConversation(in: context)
+                    return (selfUser, mlsSelfConversation)
+                }
+                guard let selfConversation else { return }
+                let isE2EICertified = try await isUserE2EICertifiedUseCase.invoke(
+                    conversation: selfConversation,
+                    user: selfUser
+                )
+                await MainActor.run {
+                    selfUserStatus.isCertified = isE2EICertified
+                }
+            } catch {
+                WireLogger.e2ei.error("failed to get E2EI certification status: \(error)")
+            }
+        }
```

Comment:
We shouldn't be writing code like this in UI layer. This can become a use case?

Reply:
I created a type `IsSelfUserE2EICertifiedUseCase` which does all the heavy lifting now.